wsd/DocumentBroker.cpp

```diff
@@ -3653,10 +3654,14 @@ std::string DocumentBroker::getJailRoot() const
 }
 
 std::size_t DocumentBroker::addSession(const std::shared_ptr<ClientSession>& session,
-                                       std::unique_ptr<WopiStorage::WOPIFileInfo> wopiFileInfo)
+                                       std::unique_ptr<WopiStorage::WOPIFileInfo> wopiFileInfo,
+                                       const std::shared_ptr<RequestVettingStation>& rvs)
 {
     try
     {
+        // Make sure we extend the life of RVS that created us,
+        // as its owner will go away (see RequestVettingStation::createClientSession()).
+        _rvs = rvs;
```

Comment:
But what does saving rvs in DocumentBroker gain us? It's _socket member is cleared at that point and its _ws member was already set as the streamSocket's handler, so the _socket member doesn't matter and it seems a little troubling to have two owners of _ws. I think I'm missing something on making the rvs live as long as the DocumentBroker.

Reply:
I'm avoiding special-handling this particular scenario. In principle, we shouldn't destroy the object from a member function, as we do now.

Agreed that DocumentBroker is not the proper owner, but what is?
Something has to own it until the hand-over is properly done.
Perhaps I need to think more about this. Suggestions welcome.

Reply:
Maybe it is sufficient to just have rvs = shared_from_this() and that captured by the setupTransfer lambda and not additionally stored in docBroker (Could name it as "selfLifecycle" like similar ones to establish that as a pattern)?

Reply:
Done.